lib/src/main/java/com/tbruyelle/rxpermissions2/RxPermissionsFragment.kt

```diff
@@ -0,0 +1,146 @@
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+
+        if (requestCode != PERMISSIONS_REQUEST_CODE) return
+
+        val shouldShowRequestPermissionRationale = BooleanArray(permissions.size)
+
+        for (i in permissions.indices) {
+            shouldShowRequestPermissionRationale[i] = shouldShowRequestPermissionRationale(permissions[i])
+        }
+
+        onRequestPermissionsResult(permissions, grantResults, shouldShowRequestPermissionRationale)
+    }
+
+    fun onRequestPermissionsResult(permissions: Array<String>, grantResults: IntArray, shouldShowRequestPermissionRationale: BooleanArray) {
+        var i = 0
+        val size = permissions.size
+        while (i < size) {
+            log("onRequestPermissionsResult  " + permissions[i])
+            // Find the corresponding subject
+            val subject = mSubjects[permissions[i]]
+            if (subject == null) {
+                // No subject found
+                Log.e(RxPermissions.TAG, "RxPermissions.onRequestPermissionsResult invoked but didn't find the corresponding permission request.")
+                return
+            }
+            mSubjects.remove(permissions[i])
+            val granted = grantResults[i] == PackageManager.PERMISSION_GRANTED
+            subject.onNext(Permission(permissions[i], granted, shouldShowRequestPermissionRationale[i]))
+            subject.onComplete()
+            i++
+        }
+    }
+
+    @TargetApi(Build.VERSION_CODES.M)
+    fun isGranted(permission: String): Boolean {
+        val fragmentActivity = activity
+                ?: throw IllegalStateException("This fragment must be attached to an activity.")
+        return fragmentActivity.checkSelfPermission(permission) == PackageManager.PERMISSION_GRANTED
+    }
+
+    @TargetApi(Build.VERSION_CODES.M)
+    fun isRevoked(permission: String): Boolean {
+        val fragmentActivity = activity
+                ?: throw IllegalStateException("This fragment must be attached to an activity.")
+        return fragmentActivity.packageManager.isPermissionRevokedByPolicy(permission, activity!!.packageName)
+    }
+
+    fun setLogging(logging: Boolean) {
+        mLogging = logging
+    }
+
+    fun getSubjectByPermission(permission: String): PublishSubject<Permission>? {
+        return mSubjects[permission]
+    }
+
+    fun containsByPermission(permission: String): Boolean {
+        return mSubjects.containsKey(permission)
+    }
+
+    fun setSubjectForPermission(permission: String, subject: PublishSubject<Permission>) {
+        mSubjects[permission] = subject
+    }
+
+    fun log(message: String) {
+        if (mLogging) {
+            Log.d(RxPermissions.TAG, message)
+        }
+    }
+
+    companion object {
+
+        private const val PERMISSIONS_REQUEST_CODE = 42
+    }
+
+    enum class SystemPermission(val permissionName: String, val action: String, val requestCode: Int, val enabledChecker: (context: Context) -> Boolean){
```

Comment:
Let's move this into a separate class so that when we import it into the Lifeworks project, it does not become RxPermissionFragment.SystemPermission